packages/airnode-validator/src/config/config.ts

```diff
@@ -153,39 +157,66 @@ export const chainOptionsSchema = z
   })
   .strict();
 
+const defaultAirnodeRrp: SuperRefinement<{
+  id?: ChainId;
+  chainId?: ChainId;
+  contracts: SchemaType<typeof airnodeRrpContractSchema>;
+}> = (chainConfig, ctx) => {
+  if (chainConfig.contracts.AirnodeRrp === '') {
+    // id OR chainId refers to the chain ID depending on the calling schema
+    const id = (chainConfig.id || chainConfig.chainId)!;
+    if (!AirnodeRrpV0[id]) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message:
+          `AirnodeRrp contract address must be specified for chain ID '${id}'` +
+          `as there was no deployment for this chain found in @airnode/protocol/deployments/references.json`,
+        path: ['contracts'],
+      });
+      return;
+    }
+    // Default to the deployed AirnodeRrp contract address if contracts is not specified
+    chainConfig.contracts = { AirnodeRrp: AirnodeRrpV0[id] };
+  }
+};
+
 export const chainAuthorizationsSchema = z.object({
   requesterEndpointAuthorizations: z.record(endpointIdSchema, z.array(evmAddressSchema)),
 });
 
 export const requesterEndpointAuthorizersSchema = z.array(evmAddressSchema);
 
-export const crossChainRequesterAuthorizerSchema = z.object({
-  requesterEndpointAuthorizers: requesterEndpointAuthorizersSchema.nonempty(),
-  chainType: chainTypeSchema,
-  chainId: z.string(),
-  contracts: airnodeRrpContractSchema,
-  chainProvider: providerSchema,
-});
+export const crossChainRequesterAuthorizerSchema = z
+  .object({
+    requesterEndpointAuthorizers: requesterEndpointAuthorizersSchema.nonempty(),
+    chainType: chainTypeSchema,
+    chainId: chainIdSchema,
+    // The type system requires optional() be transformed to the expected type despite a value being provided by superRefine
```

Comment:
Yeah, I was wondering about this in https://github.com/api3dao/airnode/pull/1755/files#r1182261043 . You should do:
1. Use `.default` to provide the default value
2. Use super refine to throw if you detect a chain for which we don't have deployment

Reply:
I tried this approach initially: `default` won't work here because the default address we're providing depends on another field, the ID of the chain, and that is only available via `superRefine`.

Reply:
I see. Then I recommend doing [transform](https://zod.dev/?id=transform) after doing the refinements.

Reply:
Do you mean not use `transform` after `.optional()` with `contracts`? Sadly that is needed too because using `optional()` alone causes problems in `airnode-node` and elsewhere because the type system thinks that the value can be undefined (doesn't know it always will be present because of `superRefine`). In other words, `transform` provides assurance that there will be a correctly typed value for the `contracts` field.

Reply:
I meant something like this https://github.com/api3dao/airnode/commit/4a752489b283cb610388d6e437c304ed049f2a3b

I believe should not use mutation inside superRefine, but you can validate during transform: https://zod.dev/?id=validating-during-transform

Reply:
Thanks for the template and link, I switched from superRefine to transform as you suggest